build/lib/release.sh

```diff
@@ -374,10 +374,14 @@ function kube::release::create_docker_images_for_server() {
         ln "${binary_file_path}" "${docker_build_path}/${binary_name}"
         ln "${KUBE_ROOT}/build/nsswitch.conf" "${docker_build_path}/nsswitch.conf"
         chmod 0644 "${docker_build_path}/nsswitch.conf"
-        cat <<EOF > "${docker_file_path}"
+        if [[ -f "${KUBE_ROOT}/build/server-image/${binary_name}/Dockerfile" ]]; then
+          ln "${KUBE_ROOT}/build/server-image/${binary_name}/Dockerfile" "${docker_file_path}"
+        else
+          cat <<EOF > "${docker_file_path}"
 FROM --platform=linux/${arch} ${base_image}
 COPY ${binary_name} /usr/local/bin/${binary_name}
 EOF
+        fi
         # ensure /etc/nsswitch.conf exists so go's resolver respects /etc/hosts
         if [[ "${base_image}" =~ busybox ]]; then
```

Comment:
Seems like this could be deleted?

Reply:
unless we support overriding the base image (I forget if we do), maybe, but probably best left to a different PR?

Reply:
I agree best left to a different PR. Thanks!